### PR TITLE
F/output cache loading

### PR DESCRIPTION
**Pull request by jcampbell05**
Implements #45.

This changes the way cert imports certificates to do the following:
1. Check if we have a valid certificate and private key in the keychain, if so use that. (Original Behaviour).
2. Check output path for valid certificate and private key, if so use that. (New Behavior).
3. Finally, try to request a new certificate, store both the new certificate and P12 in the output path and then use that. (Original Behavior modified to store the P12)

This in combination with #49, #47 and #43 will allow us to have a fully automate certificate procedure in CIs. Provided the output folder is cached between builds (Travis has the ability to a cache directories). 

In the future we could allow cert to revoke automatically certificates if we have ran out of space on the portal and generate new ones (Obviously we could give it a black list of ones not to revoke).

### CODE REVIEW

File: lib/cert/runner.rb [hunk 1]
```diff
@@ -42,6 +43,18 @@ def find_existing_cert
           Helper.log.info "Found the certificate #{certificate.id} (#{certificate.name}) which is installed on the local machine. Using this one.".green
 
           return path
+
+        elsif File.exists?(private_key_path)
```
Comment (KrauseFx) [inline lib/cert/runner.rb]: :+1:

**Comment by jcampbell05** [comment]
@KrauseFx  the tests pass fine on my local machine but they stall on Travis, any ideas?

File: lib/cert/runner.rb [hunk 2]
```diff
@@ -77,10 +90,13 @@ def create_certificate
       end
 
       # Store all that onto the filesystem
-      request_path = File.expand_path(File.join(Cert.config[:output_path], 'CertCertificateSigningRequest.certSigningRequest'))
+
+      request_path = File.expand_path(File.join(Cert.config[:output_path], "#{certificate.id}.certSigningRequest"))
```
Comment (KrauseFx) [inline lib/cert/runner.rb]: :+1:

**Comment by KrauseFx** [comment]
Thank you so much for your contribution James. Sorry it took so long :+1:

**Comment by jcampbell05** [comment]
@KrauseFx  No problem :)

**Pull request merged**